### PR TITLE
adding Logicsoft ZHC5010

### PR DESCRIPTION
**Pull request by mcnovy**

### CODE REVIEW

File: config/logicsoft/ZHC5010.xml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                </Value>
+
+                <Value type="list" genre="config" instance="1" index="26" label="Control of association groups for device 4 (button #4)" min="0" max="1" value="1" size="1" >
+                        <Help>Control of association groups for device 4 (button #4)</Help>
+                        <Item label="When commands are received by device 4, nodes in the association groups will not be switched, button activations will still send switch values." value="0" />
+                        <Item label="When commands are received by device 4, nodes in the association groups will be switched." value="1" />
+                </Value>
+
+        </CommandClass>
+<commandClass id="0085" version="2"/>
```
Comment (Fishwaldo) [inline config/logicsoft/ZHC5010.xml]: you don't need to put these other CC in the config. They should be auto discovered.

File: config/logicsoft/ZHC5010.xml [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+<commandClass id="0072" version="2"/>
+<commandClass id="0060" version="4"/>
+<commandClass id="008e" version="3"/>
+<commandClass id="0026" version="4" controlled="true"/>
+<commandClass id="0073"/>
+<commandClass id="0086" version="2"/>
+<commandClass id="005e" version="2"/>
+
+<!-- Association Groups -->
+<CommandClass id="133">
```
Comment (Fishwaldo) [inline config/logicsoft/ZHC5010.xml]: This one is fine.

**Comment by nechry** [comment]
@mcnovy do you plan to fix this PR with @Fishwaldo recommendations ?

File: config/logicsoft/ZHC5010.xml (hunk 1 above)
Comment (mcnovy) [inline config/logicsoft/ZHC5010.xml]: Thanks for taking a look at this!
Sorry for not replying, I haven't had a lot of time lately.
you mean line 171 -> 187 should just be removed?
I will do that, and test it,
though when trying to edit, I get a 404 on github at the moment, hopefully that will be resolved shortly.

**Comment by mcnovy** [comment]
@nechry  Yeah, my plan is to fix it with @Fishwaldo recommendations,
I just wan't the test it, before editing it in here, but I haven't had the time sorry.

**Comment by mcnovy** [comment]
So, I have testet Fishwaldo's changes, and they work just fine.
But, I can only change the file from my "patch 2"
Do I need to do anything else.?

**Comment by kibs** [comment]
Hi, disclaimer first: I am an Z-wave beginner :)

I have testet this on Home Assistant, in HA 5 switches are being discovered, and I am able to toggle them in HA, and the leds activates correctly in the switch. But if i press a button, the event seen by HA is not correct. 

Example: switch has node 4, and if i press the 4 buttons, the events i get in HA all have the same object_id, and HA won't register the state change.

```
INFO:homeassistant.core:Bus:Handling <Event zwave.node_event[L]: object_id=logic_soft_zhc5010_wall_switch_4, basic_level=255>
INFO:homeassistant.core:Bus:Handling <Event zwave.node_event[L]: object_id=logic_soft_zhc5010_wall_switch_4, basic_level=255>
INFO:homeassistant.core:Bus:Handling <Event zwave.node_event[L]: object_id=logic_soft_zhc5010_wall_switch_4, basic_level=255>
INFO:homeassistant.core:Bus:Handling <Event zwave.node_event[L]: object_id=logic_soft_zhc5010_wall_switch_4, basic_level=255>
```

**Comment by kibs** [comment]
and as i write this, i guess this is a problem in HA :)

**Comment by mcnovy** [comment]
Sune,

I use Home assistant as well.
I needed to setup the switches in domoticz first.

In Domoticz, go to hardware, and you z wave controller.
Select groups.

In there, you need to remove all associations from the switches, and add 1.1 to group channel 2, 7, 12, 17.

Then it should work

**Comment by kibs** [comment]
@mcnovy so I should remove node from HA and add the nodes to domoticz?

**Comment by kibs** [comment]
and also could you explain what this does, 1.1 is that a special node id?

**Comment by mcnovy** [comment]
No, the nodes are added to your z wave controller, so just leave it as it is.
But install domoticz on a computer (if on the same machine as HA, close HA first, only 1 instance can use the z wave controller at a time)
Start domoticz, and make the changes to the groups, close domoticz, and start HA, and you should be an good

**Comment by mcnovy** [comment]
Group 1.1 is just your controller, usually 1 should be fine, but the switches are just acting weird, so 1.1 is necessary

**Comment by mcnovy** [comment]
If you delete a node, it will go back to factory defaults, so don't delete them, unless you run into some issues.

**Comment by mcnovy** [comment]
Oh, and if you are going to update the firmware on the zhc switches at some time. Logicsoft offers a firmware upload utility, be advices, it will delete all nodes on your controller.

**Comment by kibs** [comment]
wow thank you, in progress updating groups as we speak, and thanks for the heads up on the updater util. wtf.. :)

**Comment by kibs** [comment]
@mcnovy It works perfectly, thank you very very much, I would never have figured that out, and also learned that the nodes was written to the controller and not in the application, and also that i can use domoticz for configuration instead of change_association in home assistant.

Thank you very much!

**Comment by mcnovy** [comment]
I've got one other remark.
If you at some point get an issue where, when you click a switch, and all your switches start switching on and off, and the only way to stop it is turning the power off, please write a comment in here.
I have seen it twice.
If it happens, start domoticz, and delete association 255 from all groups.

Not to freak you out of anything, just a heads up, and preferably a way to solve the issue fast.
Not sure it's the ZHC that is the issue though.

**Comment by kibs** [comment]
cool thanks, will do. Also now I have 1 in group 1, and 1.1 in 2,7,13,17 and nothing in the rest.

When pressing the buttons HA updates fine :) but if toggling in HA, the buttons don't turn off and HA reverts

**Comment by mcnovy** [comment]
Your welcome, we Danes don't have a lot of Danish specific z wave stuff, so Google is unfortunately not always flooded with answer's to these issues. :-)

**Comment by mcnovy** [comment]
That happens sometimes, let HA start up, and leave it for a few minutes, and try again.

**Comment by kibs** [comment]
can i find you on a chat somewhere?

**Comment by mcnovy** [comment]
Jeg har skrevet en E-Mail til dig

**Comment by skeptizzle** [comment]
Hi guys! 
I tried the solution provided by @mcnovy further up, but i'm thinking that i'm doing it wrong somehow.

1. Updated the firmware to 2.02 with another z-wave controller.
2. Added the ZHC5010 to my Hass.io (RP3 with Aeon Labs Z-Stick Gen5).
3. Switched did not work.
4. Installed Domoticz on my windows pc and used the z-wave controller that i used for the firmware update.
5. Was not able to add ZHC5010 so i factory reset the unit.
6. Adding the switch. Success.
7. Did the group thing inside Domoticz. Success.
8. Went to add the ZHC5010 to Hass.io.
9. Could not add switch.
10. After reset of the switch i could add it, but the groups where gone, of course.

The questions are, do i need to use the same controller for this operation? What is the consequence of doing this? Does this remove my z-wave network from the controller?
Any help is much appreciated! 

Cheers!

**Comment by mcnovy** [comment]
Use the same controller.
So move the controller from Hass to domoticz, do the changes, and go back. Don't make a factory default

**Comment by mcnovy** [comment]
Settings for groups etc. Is saved on the controller (zwave device), and not in domoticz or Hass.

**Comment by skeptizzle** [comment]
Hey @mcnovy. Thanks for getting back and clarifying this! I will try this when i have time and then report back! 

Cheers!

**Comment by sorenjacobi** [comment]
Hi Guys!

I just discovered how you can do it all in Home Assistant, without the need for Domoticz:

1. In Configuration / Z-Wave Node Management, find the node id of the ZHC5010 by clicking "Nodes"
1. Choose the "Services" developer tool at the bottom of the navigation.
1. For Domain, choose "zwave" and "change_association" as service. 
1. Add the first association by using this JSON code, replacing node_id with id found in Node Mangement:
{
  "association":"add",
  "node_id":8,
  "group":2,
  "target_node_id":1,
   "instance":1
}
1. Repeat the service call for all the associations by replacing the group number (2, 7, 12 & 17). 

Bonus: You can remove associations as well: 
{
  "association":"remove",
  "node_id":8,
  "group":2,
  "target_node_id":1,
  "instance":0
}

**Comment by skeptizzle** [comment]
Hi @sorenjacobi .

This is great! I will give it a try this weekend! :)

**Comment by Daigaard** [comment]
Hi All, 

I have also bought the ZHC5010 Z-Wave switch.

But I seem to be missing some steps in order to succeed with usage of this switch. Hope you can help. 
HomeAssistant: 0.59.2
ZHC5010: Firmware: 2.02 (According to dealer :-) )

I can control the switch from HA, but HA status in not updating when pushing the actual switch.  
Same problem as Kibs!!!!

I tried the solution described by: sorenjacobi

First: 
{
"association":"add",
"node_id":27,
"group":**2**,
"target_node_id":1,
"instance":1
}
{
"association":"add",
"node_id":27,
"group":**7**,
"target_node_id":1,
"instance":1
}
-----etc for the remaining 2. 

Did not solve the problem so I re-read the instruction and was doubting if I did misunderstand, so I tried:
{
"association":"add",
"node_id":27,
"group":**2**,
"target_node_id":**1**,
"instance":1
}
{
"association":"add",
"node_id":**27**,
"group":**7**,
"target_node_id":**2**,
"instance":1
}

Still no luck - Hope you guys can point me in the right direction. 

/Steffen

**Comment by Tvup** [comment]
After I followed @sorenjacobi s advice, I restarted HA. Thanks for that solution - worked perfect.
Did you try that @Daigaard ?

**Pull request merged**